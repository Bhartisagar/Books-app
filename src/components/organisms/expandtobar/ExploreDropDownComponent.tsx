import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {Box, Typography} from '@material-ui/core';
import rocket from '../../../assets/icons/rocket.png';
import science from '../../../assets/icons/science.png';
import leaf from '../../../assets/icons/leaf.png';
import globe from '../../../assets/icons/globe.png';
import bag from '../../../assets/icons/bag.png';
import man from '../../../assets/icons/man.png';
import buliding from '../../../assets/icons/buliding.png';
import health from '../../../assets/icons/health.png';
import history from '../../../assets/icons/history.png';
import bulb from '../../../assets/icons/bulb.png';
import timer from '../../../assets/icons/timer.png';
import doublecircle from '../../../assets/icons/doublecircle.png';
import statistics from '../../../assets/icons/statistics.png';
import msg from '../../../assets/icons/msg.png';
import money from '../../../assets/icons/money.png';
import heart from '../../../assets/icons/heart.png';
import education from '../../../assets/icons/education.png';
import analytical from '../../../assets/icons/analytical.png';
import { useNavigate } from 'react-router-dom';



const ExploreDropDownComponent = () => {

const useStyles = makeStyles(()=>({
  subelements:{
    paddingBottom: "1em",
    cursor:'pointer',
    
  },
  subelement1:{
    paddingBottom: "1em",
    cursor:'pointer',
    '&:hover': {
      color: 'blue'
    }
  },
  explore_container:{
    backgroundColor: "#F1F6F4",
    width:"100%",
    padding:"2em",
    display: "grid",
    justifyContent: "center",
  },
  explore_header:{
    display: "inline-flex",
    borderBottom: "1px solid black",
    paddingBottom: "1em",
    paddingRight:'1em',
    gap:'10%'
  },
  explore_titles:{
    marginRight:"4em",
  },
  explore_content:{
    paddingTop:"16px",
    paddingRight: "128px",
    display:"inline-block"
  },
  subtitles:{
    display:'flex',
    gap:'7px',
    '&:hover': {
      color: 'blue'
    }
  },
  icons:{
    height:'19px',
    width:'17px'
  }

}));
  const classes = useStyles();
  const navigate=useNavigate();
  return (
          <Box className={classes.explore_container}>
                <div className={classes.explore_header}>
                 <Typography color="primary" className={classes.explore_titles} variant="body1">Explore by category</Typography>
                  <Typography color="textPrimary" className={classes.explore_titles} variant="body1">See recently added titles</Typography>
                  <Typography color="textPrimary" className={classes.explore_titles} variant="body1">See popular titles</Typography>
                </div>
                <div className={classes.explore_content}>
                  <div className={classes.explore_content}>
                  <div className={classes.subtitles}><img src={rocket} className={classes.icons} alt="entrepreneurship"></img>   <Typography  color="textPrimary" className={classes.subelement1} variant="body2"  onClick={()=>navigate('/entrepreneurship')}>Entrepreneurship</Typography></div>
                  <div className={classes.subtitles}><img src={science} className={classes.icons} alt="science"></img> <Typography className={classes.subelements}   variant="body2">Science </Typography></div>
                  <div className={classes.subtitles}><img src={globe} className={classes.icons} alt="Economics"></img>  <Typography className={classes.subelements}   variant="body2">Economics</Typography></div>
                  <div className={classes.subtitles}><img src={bag} className={classes.icons} alt="Corporate Culture"></img>  <Typography className={classes.subelements}   variant="body2">Corporate Culture</Typography></div>
                  <div className={classes.subtitles}> <img src={man} className={classes.icons} alt="Psychology"></img>  <Typography className={classes.subelements}  variant="body2">Psychology</Typography></div>
                  <div className={classes.subtitles}> <img src={leaf} className={classes.icons} alt="Nature"></img>  <Typography className={classes.subelements}   variant="body2">Nature &amp; Environment</Typography></div>
                    
                  </div>
                  <div className={classes.explore_content}>
                  <div className={classes.subtitles}><img src={buliding} className={classes.icons} alt="Politics"></img><Typography className={classes.subelements}  variant="body2">Politics</Typography> </div>
                  <div className={classes.subtitles}><img src={health} className={classes.icons} alt="health"></img> <Typography className={classes.subelements}  variant="body2">Health &amp; Nutrition</Typography> </div>
                  <div className={classes.subtitles}><img src={history} className={classes.icons} alt="history"></img>    <Typography className={classes.subelements}  variant="body2">History</Typography> </div>
                  <div className={classes.subtitles}><img src={bulb} className={classes.icons} alt="Motivation"></img>    <Typography className={classes.subelements}   variant="body2">Motivation &amp; Inspiration</Typography> </div>
                  <div className={classes.subtitles}><img src={timer} className={classes.icons} alt="Productivity"></img>    <Typography className={classes.subelements}   variant="body2">Productivity</Typography> </div>
                  <div className={classes.subtitles}><img src={doublecircle} className={classes.icons} alt="Career"></img>    <Typography className={classes.subelements}  variant="body2">Career &amp; Success</Typography> </div>

                   </div>
                  <div className={classes.explore_content}>
                  <div className={classes.subtitles}><img src={statistics} className={classes.icons} alt="Marketing"></img>   <Typography className={classes.subelements}   variant="body2">Marketing &amp; Sales</Typography> </div>
                  <div className={classes.subtitles}><img src={analytical} className={classes.icons} alt=" Personal Development"></img>   <Typography className={classes.subelements}   variant="body2">Personal Development</Typography> </div>
                  <div className={classes.subtitles}><img src={msg} className={classes.icons} alt=" Communication Skills"></img>    <Typography className={classes.subelements}   variant="body2">Communication Skills</Typography> </div>
                  <div className={classes.subtitles}><img src={money} className={classes.icons} alt="Investments "></img>    <Typography className={classes.subelements}   variant="body2">Money &amp; Investments</Typography> </div>
                  <div className={classes.subtitles}><img src={heart} className={classes.icons} alt="Relationships "></img>    <Typography className={classes.subelements}   variant="body2">Sex &amp; Relationships</Typography> </div>
                  <div className={classes.subtitles}><img src={education} className={classes.icons} alt="Education"></img>    <Typography className={classes.subelements}   variant="body2">Education</Typography> </div>
                 
                 </div>
                </div>
              </Box>
  );
};

export default ExploreDropDownComponent;
