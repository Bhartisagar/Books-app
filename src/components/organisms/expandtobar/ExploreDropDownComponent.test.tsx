import { render, screen } from "@testing-library/react";
import ExploreDropDownComponent from "./ExploreDropDownComponent";
import { MemoryRouter } from "react-router";
import renderer from 'react-test-renderer'

test("first snapshot",()=>{
    const component = renderer.create(
        <MemoryRouter><ExploreDropDownComponent/></MemoryRouter>
    )
    let tree=component.toJSON();

    expect(tree).toMatchSnapshot();
})


test('should render entrepreneur icon',()=>{
      render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
      const Icon = screen.getByRole('heading',{name:"Explore by category"})
      expect(Icon).toBeInTheDocument();
});
test('should render heading See recently added titles',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByRole('heading',{name:"See recently added titles"})
    expect(Icon).toBeInTheDocument();
});
test('should render See popular titles icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByRole('heading',{name:"See popular titles"})
    expect(Icon).toBeInTheDocument();
});
test('should render rocket icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('entrepreneurship')
    expect(Icon).toBeInTheDocument();
});
test('should render health icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('health')
    expect(Icon).toBeInTheDocument();
});
test('should render science icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('science')
    expect(Icon).toBeInTheDocument();
});
test('should render Econics icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Economics')
    expect(Icon).toBeInTheDocument();
});
test('should render Motivation icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Motivation')
    expect(Icon).toBeInTheDocument();
});
test('should render Productivity icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Productivity')
    expect(Icon).toBeInTheDocument();
});
test('should render Career icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Career')
    expect(Icon).toBeInTheDocument();
});
test('should render Personal Development icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Personal Development')
    expect(Icon).toBeInTheDocument();
});
test('should render  icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Communication Skills')
    expect(Icon).toBeInTheDocument();
});
test('should render Investments icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Investments')
    expect(Icon).toBeInTheDocument();
});
test('should render Relationships icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Relationships')
    expect(Icon).toBeInTheDocument();
});
test('should render Education icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Education')
    expect(Icon).toBeInTheDocument();
});
test('should render history icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('history')
    expect(Icon).toBeInTheDocument();
});
test('should render Corporate Culture icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Corporate Culture')
    expect(Icon).toBeInTheDocument();
});
test('should render Psychology icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Psychology')
    expect(Icon).toBeInTheDocument();
});
test('should render Nature icon',()=>{
    render(<MemoryRouter><ExploreDropDownComponent/></MemoryRouter>)
    const Icon = screen.getByAltText('Nature')
    expect(Icon).toBeInTheDocument();
});