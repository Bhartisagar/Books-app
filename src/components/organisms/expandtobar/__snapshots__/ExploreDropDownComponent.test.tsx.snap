// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-9 makeStyles-explore_container-3"
>
  <div
    className="makeStyles-explore_header-4"
  >
    <p
      className="MuiTypography-root makeStyles-explore_titles-5 MuiTypography-body1 MuiTypography-colorPrimary"
    >
      Explore by category
    </p>
    <p
      className="MuiTypography-root makeStyles-explore_titles-5 MuiTypography-body1 MuiTypography-colorTextPrimary"
    >
      See recently added titles
    </p>
    <p
      className="MuiTypography-root makeStyles-explore_titles-5 MuiTypography-body1 MuiTypography-colorTextPrimary"
    >
      See popular titles
    </p>
  </div>
  <div
    className="makeStyles-explore_content-6"
  >
    <div
      className="makeStyles-explore_content-6"
    >
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="entrepreneurship"
          className="makeStyles-icons-8"
          src="rocket.png"
        />
           
        <p
          className="MuiTypography-root makeStyles-subelement1-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
          onClick={[Function]}
        >
          Entrepreneurship
        </p>
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="science"
          className="makeStyles-icons-8"
          src="science.png"
        />
         
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Science 
        </p>
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Economics"
          className="makeStyles-icons-8"
          src="globe.png"
        />
          
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Economics
        </p>
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Corporate Culture"
          className="makeStyles-icons-8"
          src="bag.png"
        />
          
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Corporate Culture
        </p>
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
         
        <img
          alt="Psychology"
          className="makeStyles-icons-8"
          src="man.png"
        />
          
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Psychology
        </p>
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
         
        <img
          alt="Nature"
          className="makeStyles-icons-8"
          src="leaf.png"
        />
          
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Nature & Environment
        </p>
      </div>
    </div>
    <div
      className="makeStyles-explore_content-6"
    >
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Politics"
          className="makeStyles-icons-8"
          src="buliding.png"
        />
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Politics
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="health"
          className="makeStyles-icons-8"
          src="health.png"
        />
         
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Health & Nutrition
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="history"
          className="makeStyles-icons-8"
          src="history.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          History
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Motivation"
          className="makeStyles-icons-8"
          src="bulb.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Motivation & Inspiration
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Productivity"
          className="makeStyles-icons-8"
          src="timer.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Productivity
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Career"
          className="makeStyles-icons-8"
          src="doublecircle.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Career & Success
        </p>
         
      </div>
    </div>
    <div
      className="makeStyles-explore_content-6"
    >
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Marketing"
          className="makeStyles-icons-8"
          src="statistics.png"
        />
           
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Marketing & Sales
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt=" Personal Development"
          className="makeStyles-icons-8"
          src="analytical.png"
        />
           
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Personal Development
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt=" Communication Skills"
          className="makeStyles-icons-8"
          src="msg.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Communication Skills
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Investments "
          className="makeStyles-icons-8"
          src="money.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Money & Investments
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Relationships "
          className="makeStyles-icons-8"
          src="heart.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Sex & Relationships
        </p>
         
      </div>
      <div
        className="makeStyles-subtitles-7"
      >
        <img
          alt="Education"
          className="makeStyles-icons-8"
          src="education.png"
        />
            
        <p
          className="MuiTypography-root makeStyles-subelements-1 MuiTypography-body2"
        >
          Education
        </p>
         
      </div>
    </div>
  </div>
</div>
`;
